### PR TITLE
feat(server): create a person with optional values

### PR DESCRIPTION
**Pull request by jrasm91**
* speed up person e2e (remove beforeEach)
* add option to set `birthDate`, `isHidden`, and `name` when creating a person
* sort person controller methods
* rename person service methods
* refactor person service update method to have less branching

### CODE REVIEW

File: e2e/src/api/specs/person.e2e-spec.ts [hunk 1]
```diff
@@ -141,6 +145,41 @@ describe('/activity', () => {
...
 
+  describe('POST /person', () => {
+    it('should require authentication', async () => {
+      const { status, body } = await request(app).post(`/person`);
+      expect(status).toBe(401);
+      expect(body).toEqual(errorDto.unauthorized);
+    });
+
+    it('should not accept invalid birth dates', async () => {
+      for (const { birthDate, response } of invalidBirthday) {
```
Comment (danieldietzler) [inline e2e/src/api/specs/person.e2e-spec.ts:156]: Maybe we should move the for each out of the `it` for a more helpful output if the test fails?

File: server/src/domain/person/person.service.ts [hunk 2]
```diff
@@ -199,36 +196,41 @@ export class PersonService {
...
 
-      person = await this.repository.update({ id, faceAssetId: face.id });
+      faceId = face.id;
+    }
+
+    const person = await this.repository.update({ id, faceAssetId: faceId, name, birthDate, isHidden });
+
+    if (assetId) {
       await this.jobRepository.queue({ name: JobName.GENERATE_PERSON_THUMBNAIL, data: { id } });
     }
```
Comment (danieldietzler) [inline server/src/domain/person/person.service.ts:228]: Can't we merge this if with the if starting in l. 214?

**Review by danieldietzler — APPROVED** [review]
LGTM

File: server/src/domain/person/person.service.ts (hunk 2 above)
Comment (jrasm91) [reply server/src/domain/person/person.service.ts:228]: No, the faceId needs to be saved before the job can be queued. I don't want there to be a race condition if the job runs before the update happens.

File: e2e/src/api/specs/person.e2e-spec.ts (hunk 1 above)
Comment (jrasm91) [reply e2e/src/api/specs/person.e2e-spec.ts:156]: I will do it in a follow up. There is only one test that runs either way right now.

**Pull request merged**